superflore/utils.py

```diff
@@ -286,3 +286,18 @@ def url_to_repo_org(url):
         )
     url = url.replace('https://github.com/', '').split('/')
     return url[0], url[1]
+
+
+def retry_on_exception(callback, *args, retries=5, retry_msg='', error_msg=''):
+    for retry in range(1, retries+1):
+        try:
+            ret = callback(*args)
+        except Exception as e:
+            if retry == retries:
+                if error_msg:
+                    err(f'{error_msg} {e} {retry}/{retries}')
+                raise e
+            if retry_msg:
+                warn(f'{retry_msg} {e} {retry}/{retries}...')
+        else:
+            return ret
```

Comment:
It would be good to at least warn possibly error if the retry count is out of a valid range, (negative or zero) as the callback will never be tried.

It would probably make more sense to actually have it hit the callback first and then count retries. So an argument "retries=0" will still imply it tries once and then quits, which is 0 retries.

Reply:
Thanks @tfoote << Ptal